src/monster.cpp

```diff
@@ -2139,6 +2153,37 @@ void monster::die( Creature *nkiller )
     }
 }
 
+bool monster::mod_mech_power( int amt )
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    amt = -amt;
+    battery_item->ammo_consume( amt, pos() );
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    return true;
+}
+
+bool monster::check_mech_powered()
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    itype_id mech_bat = itype_id( type->mech_battery );
+    const itype &type = *item::find_type( mech_bat );
+    float low_charge = ( static_cast<float>( type.magazine->capacity ) / 100 ) * 10;
+    if( battery_item->ammo_remaining() <= low_charge && one_in( 10 ) ) {
```

Comment:
Making `low_charge` a float is a bit pointless as `ammo_remaining` returns an int anyway. Comparing for example with 10.1f is same as comparing with 10.9f.